langchain/llms/accelerate.py

```diff
@@ -0,0 +1,148 @@
+# TODO Garbage collect loaded models on exit?!?
+# TODO Get Contextsizes for model IDs to impleme max_length=-1
+# TODO Implement bf16 and model nativ types
+# TODO Update gitignore or find better place to save model_weights
+
+import os
+from typing import Any, List, Mapping, Optional
+
+from pydantic import BaseModel, Extra
+from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer, pipeline
+
+from langchain.llms.base import LLM, BaseLLM
+from langchain.llms.utils import enforce_stop_tokens
+
+DEFAULT_MODEL_NAME = "EleutherAI/gpt-neox-20b"  # nme as of Huggingface
+
+
+class Accelerate(LLM, BaseModel):
+    """Interference on big models on consumer hardware with the help off
+    accelerate system. It distributes models that dont fit in video ram over multiple gpus
+    and system ram, and if needed to disk (not yet implemented).
+    A locl copy of the weights for much faster loading than hugging face standard is created.
+
+    Tested with
+    "EleutherAI/gpt-neox-20b", "EleutherAI/gpt-j-6B", "facebook/opt-30b"
+
+    30b model needs ~64gb = RAM+VRM and on an RTX 3080 16GB and 64 GB RAM it takes 10 Sek per token.
+
+    Example:
+        .. code-block:: python
+
+            from langchain.llms import Accelerate
+            model_name = "facebook/opt-30b"
+            FastLLM = Accelerate.from_model_name(model_name=model_name)
+            print(FastLLM("Hello World"))
+
+    """
+
+    model_name: str = DEFAULT_MODEL_NAME
+    """Model name to use."""
+    tokenizer: Any = None
+    """Tokenizer to use."""
+    model: Any = None
+    """model to use."""
+
+    class Config:
+        """Configuration for this pydantic object."""
+
+        extra = Extra.forbid
+
+    @staticmethod
+    def get_accelerated_gpt(model_name: str = DEFAULT_MODEL_NAME) -> Any:
+        import torch
+        from accelerate import (
+            infer_auto_device_map,
+            init_empty_weights,
+            load_checkpoint_and_dispatch,
+        )
+        from transformers import AutoConfig, AutoModelForCausalLM, AutoTokenizer
+
+        # what else models are available and compatible? TODO
+
+        weights_path = "./accelerated_model_weights/" + model_name
+        if not os.path.exists(weights_path):
+            model = AutoModelForCausalLM.from_pretrained(
+                model_name, torch_dtype=torch.float16
+            )
+            os.makedirs(weights_path)
+            # here an even faster method for saving and loading is described
+            # https://www.philschmid.de/deploy-gptj-sagemaker
+            model.save_pretrained(weights_path)
+        config = AutoConfig.from_pretrained(model_name)
+
+        config.use_cache = False

```

Comment:
why disable HF cache use and create a new cache path manually as you are doing here?